components/script_bindings/codegen/CodegenRust.py

```diff
@@ -7566,7 +7566,7 @@ def __init__(self, config, prefix, webIDLFile):
         cgthings += [CGGeneric(
             f"pub(crate) type {t.identifier.name} = "
             f"{originalBinding}::{t.identifier.name}"
-            f"{'::<crate::DomTypeHolder>' if containsDomInterface(t.innerType) else ''};"
+            f"{'::<crate::DomTypeHolder>' if containsDomInterface(t.innerType) or t.innerType.isPromise() else ''};"
```

Comment:
thought: Should we make `containsDomInterface` return true if promise?

Reply:
That was my suggestion in #36083

Reply:
> I tried adding `or t.isPromise()` both at lines 77 and 91, and compilation failed both times with `AttributeError: IDLDictionary object has no attribute isPromise`

maybe we need something like `isinstance(t, IDLType) and t.isPromise()`